share/completions/terraform.fish

```diff
@@ -0,0 +1,171 @@
+# general options
+complete -f -c terraform -l version -d 'Print version information'
+complete -f -c terraform -l help -d 'Show help'
+
+### apply
+complete -f -c terraform -n '__fish_use_subcommand' -a apply -d 'Builds or changes infrastructure'
```

Comment:
Done for high-level commands. Just so you know, the descriptions were based directly on `terraform --help`.